api/v1/helper.go

```diff
@@ -801,3 +802,39 @@ func NetFilterMatch(netFilter string, netValue string) (isMatch bool) {
 
 	return netFilterResult[0][1] == netValueResult[0][1] && netFilterResult[0][2] == netValueResult[0][2]
 }
+
+// MaxUnavailable calculate the max number of unavailable nodes to represent the number of nodes
+// we can drain in parallel
+func (s *SriovNetworkPoolConfig) MaxUnavailable(numOfNodes int) (int, error) {
+	// this means we want to drain all the nodes in parallel
+	if s.Spec.MaxUnavailable == nil {
+		return -1, nil
+	}
+	intOrPercent := *s.Spec.MaxUnavailable
+
+	if intOrPercent.Type == intstrutil.String {
+		if strings.HasSuffix(intOrPercent.StrVal, "%") {
+			i := strings.TrimSuffix(intOrPercent.StrVal, "%")
+			v, err := strconv.Atoi(i)
+			if err != nil {
+				return 0, fmt.Errorf("invalid value %q: %v", intOrPercent.StrVal, err)
+			}
+			if v > 100 || v < 1 {
+				return 0, fmt.Errorf("invalid value: percentage needs to be between 1 and 100")
+			}
+		} else {
+			return 0, fmt.Errorf("invalid type: strings needs to be a percentage")
```

Comment:
`HasSuffix` is not enough to detect percentage.

```
func isPercentage(input string) bool {
	pattern := `^(\d+(\.\d+)?|\.\d+)%$`

	regex, err := regexp.Compile(pattern)
	if err != nil {
		fmt.Println("Error compiling regular expression:", err)
		return false
	}

	return regex.MatchString(input)
}
```

Reply:
Looking below, I think we shouldn't even do all kinds of logic on check if it is between 0 and 100 percent. I think we should instead just call GetScaledValueFromIntOrPercent and let it deal with erroring out if it's incorrect. After that function call, we can check if the value is between 1 and max nodes. Note also that your code is using 1% which can be less than 1 node.